src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.cpp

```diff
@@ -124,6 +132,13 @@ void ShenandoahBarrierSet::on_thread_detach(Thread *thread) {
       gclab->retire();
     }
 
+    if (ShenandoahCardBarrier) {
```

Comment:
Fairly weird to see this conditionalized on `ShenandoahCardBarrier`. This seems to be a crude check for generational. But I think this is unnecessary, since PLAB would be null if generational is disabled?

Reply:
Much of the code here uses specific barrier flags like this (`ShenandoahStackWatermarkBarrier`, is a proxy for SATB mode, for example). But you are right, the PLAB should be null in non-generational modes and the accessor for the generational heap will assert if we try to use it here outside of non-generational modes.

Reply:
https://bugs.openjdk.org/browse/JDK-8340395